apis/generators/v1alpha1/generator_vault.go

```diff
@@ -0,0 +1,46 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type VaultDynamicSecretSpec struct {
+	// Vault provider common spec
+	Vault *esv1beta1.VaultProvider `json:"vault"`
```

Comment:
It sounds weird to have a `spec.vault` field in the `Kind=VaultDynamicSecret`, IMO `spec.provider` may be a better name for it.